### ISSUE TITLE
TSLint CLI exists with 0 code instead of 2 when there are linting errors

### ISSUE THREAD

**Issue by rodrigomf24** [palantir/tslint#4508]
### Bug Report

- __TSLint version__: 5.11.0
- __TypeScript version__: 3.0.3
- __Running TSLint via__: (pick one) CLI

#### TypeScript code being linted

```ts
// code snippet
export const someGeneratorFunction = function* ( ) {
  console.log( "this should not be here    " );
  console.log( "this should not be here 22222   " );
}
```

with `tslint.json` configuration:

```json
{
  "compilerOptions": {
    "strict": true,
    "alwaysStrict": true,
    "target": "es2015",
    "module": "es2015",
    "lib": [
      "dom",
      "es2017"
    ],
    "moduleResolution": "node",
    "rootDir": "src",
    "baseUrl": ".",
    "outDir": "artifacts/dist",
    "sourceMap": true,
    "allowJs": false,
    "allowSyntheticDefaultImports": true,
    "forceConsistentCasingInFileNames": true,
    "allowUnreachableCode": false,
    "declaration": false,
    "experimentalDecorators": true,
    "emitDecoratorMetadata": true,
    "noUnusedLocals": true,
    "noUnusedParameters": true,
    "jsx": "react"
  },
  "include": [
    "src"
  ],
  "exclude": [
    "node_modules",
    "src/setupTests.ts",
    "**/__tests__",
    "src/__test_settings__"
  ],
  "rules": {
    "no-console": [
      true,
      "log"
    ]
  }
}

```

#### Actual behavior

Running the command does returns through stdout the list of linting errors, but the exit code is not the correct one(2), it returns 0. The reason I can confirm it is not returning 0 is because the npm command that executes this `./node_modules/.bin/tslint src/**/*.ts{,x} -c precommit.tsconfig.json --fix` doesn't fail, I've even tested running it through a bash script and validated if the command succeeded, and it returned exit code 0 instead of 2.

<img width="1641" alt="tslint" src="https://user-images.githubusercontent.com/3925338/52317775-4b6fab00-2976-11e9-9b22-9a0905df232e.png">

#### Expected behavior

Should return exit code 2 as stated on the CLI documentation if there are linting errors.

**Comment by pronebird** [comment]
I wonder if setting a `defaultSeverity` to `error` has any impact on the exit code?

**Comment by adidahiya** [comment]
It's a little strange that your tsconfig.json and tslint.json are the same file. Have you tried separating them?

**Comment by JoshuaKGoldberg** [comment]
☠️ *TSLint's time has come!* ☠️ 

TSLint is no longer accepting most feature requests per #4534. See **[typescript-eslint.io](https://typescript-eslint.io)** for the new, shiny way to lint your TypeScript code with **ESLint**. ✨ 

It was a pleasure open sourcing with you all!

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
